### PR TITLE
Add paid events

### PR DESCRIPTION
**Pull request by kolotum**
## Description

This PR adds support for paid events in iOS and Android.

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/google_mobile_ads/lib/src/ad_instance_manager.dart [hunk 1]
```diff
@@ -96,6 +96,14 @@ class AdInstanceManager {
         assert(arguments['rewardItem'] != null);
         ad.listener?.onRewardedAdUserEarnedReward(ad, arguments['rewardItem']);
         break;
+      case 'onPaidEvent':
+        ad.listener?.onPaidEvent(
+          ad,
+          arguments['valueMicros'] / 1000000,
```
Comment (blasten) [inline packages/google_mobile_ads/lib/src/ad_instance_manager.dart:102]: Maybe assert the type before reading

Comment (blasten) [reply packages/google_mobile_ads/lib/src/ad_instance_manager.dart:102]: Also, this needs a test. https://github.com/googleads/googleads-mobile-flutter/blob/99ab94d8ea46aa3abd8dd39ce9e49feade736d70/packages/google_mobile_ads/test/ad_containers_test.dart#L155 has an example about how to test the calls made to the channel.

Comment (kolotum) [reply packages/google_mobile_ads/lib/src/ad_instance_manager.dart:103]: @blasten I have added a test similiar to onRewardedAdUserEarnedReward

File: packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package io.flutter.plugins.googlemobileads;
+
+import androidx.annotation.NonNull;
+
+public class FlutterAdValue {
+    @NonNull final int precisionType;
```
Comment (blasten) [inline packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java:6]: `@NonNull` isn't used for primitive types (int and double)

Comment (blasten) [reply packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java:6]: below too

Comment (kolotum) [reply packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java]: changed

**Comment by blasten** [comment]
@zoeyfan do you have any thoughts on this feature?

File: packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package io.flutter.plugins.googlemobileads;
+
+import androidx.annotation.NonNull;
+
+public class FlutterAdValue {
```
Comment (blasten) [inline packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdValue.java:5]: nit: java doc

**Review by blasten — APPROVED** [review]
LGTM + nit

**Comment by jjliu15** [comment]
Closing in favor of #243

**Pull request closed without merging**